diff --git a/core/options.cpp b/core/options.cpp
index cf704c2..bebc272 100644
--- a/core/options.cpp
+++ b/core/options.cpp
@@ -418,7 +418,7 @@ bool OptsInternal::Parse(boost::program_options::variables_map &vm, RPiCamApp *a
 	std::vector<std::shared_ptr<libcamera::Camera>> cameras = app->GetCameras();
 	if (camera < cameras.size())
 	{
-		const std::string cam_id = *cameras[camera]->properties().get(libcamera::properties::Model);
+		const std::string cam_id = std::string(*cameras[camera]->properties().get(libcamera::properties::Model));
 
 		if (cam_id.find("imx708") != std::string::npos)
 		{
diff --git a/core/rpicam_app.cpp b/core/rpicam_app.cpp
index 9e5a2d0..fa402b4 100644
--- a/core/rpicam_app.cpp
+++ b/core/rpicam_app.cpp
@@ -129,7 +129,7 @@ std::string const &RPiCamApp::CameraId() const
 std::string RPiCamApp::CameraModel() const
 {
 	auto model = camera_->properties().get(properties::Model);
-	return model ? *model : camera_->id();
+	return model ? std::string(*model) : std::string(camera_->id());
 }
 
 void RPiCamApp::OpenCamera()
